lib/Dialect/Arc/Transforms/InferStateProperties.cpp

```diff
@@ -422,56 +409,56 @@ void InferStatePropertiesPass::runOnStateOp(
     return;
 
   auto outputOp = cast<arc::OutputOp>(arc.getBodyBlock().getTerminator());
-  const unsigned visitedNoChange = -1;
-
-  // Check for reset patterns, we only have to do this once per arc::DefineOp
-  // and store the result for later arc::StateOps referring to the same arc.
-  if (!resetConditionMap.count(arc)) {
-    SmallVector<ResetInfo> resetInfos;
-    int numResets = 0;
-    ;
-    for (auto &output : outputOp->getOpOperands()) {
-      auto resetInfo = computeResetInfoFromPattern(output);
-      resetInfos.push_back(resetInfo);
-      if (resetInfo)
-        ++numResets;
-    }
-
-    // Rewrite the arc::DefineOp if valid
-    auto result = applyResetTransformation(arc, resetInfos);
-    if ((succeeded(result) && resetInfos[0]))
-      resetConditionMap[arc] = resetInfos[0].condition.getArgNumber();
-    else
-      resetConditionMap[arc] = visitedNoChange;
+  static constexpr unsigned visitedNoChange = -1;
+
+  if (detectResets) {
+    // Check for reset patterns, we only have to do this once per arc::DefineOp
+    // and store the result for later arc::StateOps referring to the same arc.
+    if (!resetConditionMap.count(arc)) {
+      SmallVector<ResetInfo> resetInfos;
+      int numResets = 0;
+      ;
```

Comment:
Not an issue of this PR, but what is this lonely semicolon doing here? 😆

Reply:
Let me get rid of that :rofl:

Reply:
```suggestion
```